lib/collection/src/grouping/aggregator.rs

```diff
@@ -0,0 +1,332 @@
+use std::collections::{HashMap, HashSet};
+
+use segment::types::{ExtendedPointId, ScoredPoint};
+
+type Hits = HashSet<ScoredPoint>;
+
+#[allow(dead_code)] // temporary
+struct GroupsAggregator {
+    groups: HashMap<String, Hits>,
+    max_group_size: usize,
+    grouped_by: String,
+    max_groups: usize,
+}
+
+#[allow(dead_code)] // temporary
+impl GroupsAggregator {
+    fn new(groups: usize, group_size: usize, grouped_by: String) -> Self {
+        Self {
+            groups: HashMap::with_capacity(groups),
+            max_group_size: group_size,
+            grouped_by,
+            max_groups: groups,
+        }
+    }
+
+    /// Adds a point to the group that corresponds based on the group_by field, assumes that the point has the group_by field
+    fn add_point(&mut self, point: &ScoredPoint) {
+        // if the key contains multiple values, grabs the first one
+        let group_value = point
+            .payload
+            .as_ref()
+            .unwrap()
+            .get_value(&self.grouped_by)
+            .values()
+            .first()
+            .unwrap()
+            .to_string()
```

Comment:
We want to use it for generating the key on the hashmap, that's why I am using string. But now that I think about it, we should be able to use the actual Value enum. One of the problems I had is that the `Display` was inserting literal `/"` characters on the string, so it may be a better option

Reply:
After trying it I realized `serde_json::Value` does not implement `Hash`, so one solution is to filter out `Value::Object` and `Value::Array`, and keep using the `Display` representation as key, wdyt?

Reply:
Implemented a new wrapper struct for this purpose: `GroupKey`